statistics/cmsketch.go

```diff
@@ -670,12 +670,14 @@ func MergeTopN(topNs []*TopN, n uint32) (*TopN, []TopNMeta) {
 	}
 
 	var finalTopN TopN
+	remain := n
 	finalTopN.TopN = make([]TopNMeta, 0, n)
 	popedTopNPair := make([]TopNMeta, 0, uint32(numTop)-n)
 	for value, cnt := range counter {
 		data := hack.Slice(string(value))
-		if n > 0 && cnt >= lastTopCnt {
+		if n > 0 && remain > 0 && cnt >= lastTopCnt {
```

Comment:
The condition `remain > 0` may make the result unstable when multiple items have the same `lastTopCnt`.
Could we also update the sort algorithm above to make it stable?